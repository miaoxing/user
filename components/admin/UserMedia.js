import { Component } from 'react';
import {Popover, Descriptions} from 'antd';
import {Box} from '@mxjs/box';
import Media from '@mxjs/a-media';
import {ManOutlined, WomanOutlined, CheckCircleTwoTone} from '@ant-design/icons';
import PropTypes from 'prop-types';
import {Avatar} from '@miaoxing/admin';

export default class extends Component {
  static propTypes = {
    user: PropTypes.shape({
      sex: PropTypes.number,
      name: PropTypes.string,
      nickName: PropTypes.string,
      displayName: PropTypes.string,
      avatar: PropTypes.string,
      mobile: PropTypes.string,
      isMobileVerified: PropTypes.bool,
      country: PropTypes.string,
      province: PropTypes.string,
      city: PropTypes.string,
    }).isRequired,
  };

  renderSex() {
    switch (this.props.user.sex) {
      case 1:
        return <ManOutlined style={{color: '#0c86de'}}/>;

      case 2:
        return <WomanOutlined style={{color: '#f28c48'}}/>;

      default:
        return '';
    }
  }

  render() {
    const user = this.props.user;
    return (
      <Media>
        <Popover
          placement="rightTop"
          trigger="hover"
          content={<>
            <Media>
              <Avatar user={user} shape="square" size={96}/>
              <Media.Body css={{width: 320}}>
                <Descriptions title={
                  <>
                    {user.nickName}
                    {' '}
                    {this.renderSex()}
                  </>
                } column={1}>
                  <Descriptions.Item label="姓名">{user.name}</Descriptions.Item>
                  <Descriptions.Item label="手机">
                    {user.mobile || '-'}
                    {' '}
                    {user.isMobileVerified ? <Box ml1><CheckCircleTwoTone twoToneColor="#52c41a"/></Box> : ''}
                  </Descriptions.Item>
                  <Descriptions.Item label="地区">
                    {[user.country, user.province, user.city].filter(el => el).join(' ') || '-'}
                  </Descriptions.Item>
                </Descriptions>
              </Media.Body>
            </Media>
          </>}>
          <Avatar user={user} shape="square" size={48}/>
        </Popover>
        <Media.Body>
          {user.displayName}
        </Media.Body>
      </Media>
    );
  }
}
