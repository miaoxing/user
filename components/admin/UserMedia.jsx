import {Popover, Descriptions} from 'antd';
import Media from '@mxjs/a-media';
import {ManOutlined, WomanOutlined, CheckCircleTwoTone} from '@ant-design/icons';
import PropTypes from 'prop-types';
import {Avatar} from '@miaoxing/admin';

const UserMedia = ({user, body}) => {
  return (
    <Media>
      <Popover
        placement="rightTop"
        trigger="hover"
        content={<>
          <Media>
            <Avatar user={user} shape="square" size={96}/>
            <Media.Body className="w-80">
              <Descriptions title={
                <>
                  {user.nickName}
                  {' '}
                  {user.sex === 1 && <ManOutlined style={{color: '#0c86de'}}/>}
                  {user.sex === 2 && <WomanOutlined style={{color: '#f28c48'}}/>}
                </>
              } column={1}>
                <Descriptions.Item label="姓名">{user.name}</Descriptions.Item>
                <Descriptions.Item label="手机">
                  {user.mobile || '-'}
                  {' '}
                  {user.isMobileVerified ? <CheckCircleTwoTone className="ml-1 self-center" twoToneColor="#52c41a"/> : ''}
                </Descriptions.Item>
                <Descriptions.Item label="地区">
                  {[user.country, user.province, user.city].filter(el => el).join(' ') || '-'}
                </Descriptions.Item>
              </Descriptions>
            </Media.Body>
          </Media>
        </>}>
        <Avatar user={user} shape="square" size={54}/>
      </Popover>
      <Media.Body>
        {user.displayName}
        {body && <div className="text-gray-500">{body}</div>}
      </Media.Body>
    </Media>
  );
};

UserMedia.propTypes = {
  user: PropTypes.shape({
    sex: PropTypes.number,
    name: PropTypes.string,
    nickName: PropTypes.string,
    displayName: PropTypes.string,
    avatar: PropTypes.string,
    mobile: PropTypes.string,
    isMobileVerified: PropTypes.bool,
    country: PropTypes.string,
    province: PropTypes.string,
    city: PropTypes.string,
  }).isRequired,
  body: PropTypes.node,
};

export default UserMedia;
