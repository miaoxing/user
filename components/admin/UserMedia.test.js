import UserMedia from './UserMedia';
import {render, fireEvent} from '@testing-library/react';

describe('UserMedia', () => {
  test('basic', () => {
    const result = render(<UserMedia user={{}}/>);
    expect(result.container).toMatchSnapshot();
  });

  test('user', async () => {
    const {container, ...result} = render(<UserMedia user={{
      sex: 1,
      name: 'name',
      nickName: 'nickName',
      displayName: 'displayName',
      avatar: 'avatar',
      mobile: 'mobile',
      isMobileVerified: true,
      country: 'country',
      province: 'province',
      city: 'city',
    }}/>);
    expect(container).toMatchSnapshot();

    fireEvent.mouseOver(container.querySelector('img'));
    await result.findByText('姓名');
    expect(container).toMatchSnapshot();
    expect(result.baseElement.querySelector('.ant-popover')).toMatchSnapshot();
  });
});
