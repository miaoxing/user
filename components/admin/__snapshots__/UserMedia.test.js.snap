// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMedia basic 1`] = `
<div>
  <div
    class="media"
  >
    <img
      class="chakra-image__placeholder css-1ewqzxf"
    />
    <div
      class="media-body"
    />
  </div>
</div>
`;

exports[`UserMedia user 1`] = `
<div>
  <div
    class="media"
  >
    <img
      class="chakra-image__placeholder css-1ewqzxf"
    />
    <div
      class="media-body"
    >
      displayName
    </div>
  </div>
</div>
`;

exports[`UserMedia user 2`] = `
<div>
  <div
    class="media"
  >
    <img
      class="ant-popover-open css-1ewqzxf"
    />
    <div
      class="media-body"
    >
      displayName
    </div>
  </div>
</div>
`;

exports[`UserMedia user 3`] = `
<div
  class="ant-popover"
  style="opacity: 0; pointer-events: none;"
>
  <div
    class="ant-popover-content"
  >
    <div
      class="ant-popover-arrow"
    >
      <span
        class="ant-popover-arrow-content"
      />
    </div>
    <div
      class="ant-popover-inner"
      role="tooltip"
    >
      <div
        class="ant-popover-inner-content"
      >
        <div
          class="media"
        >
          <img
            class="chakra-image__placeholder css-115kvau"
          />
          <div
            class="css-ddbaxo media-body"
          >
            <div
              class="ant-descriptions"
            >
              <div
                class="ant-descriptions-header"
              >
                <div
                  class="ant-descriptions-title"
                >
                  nickName
                   
                  <span
                    aria-label="man"
                    class="anticon anticon-man"
                    role="img"
                    style="color: rgb(12, 134, 222);"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="man"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M874 120H622c-3.3 0-6 2.7-6 6v56c0 3.3 2.7 6 6 6h160.4L583.1 387.3c-50-38.5-111-59.3-175.1-59.3-76.9 0-149.3 30-203.6 84.4S120 539.1 120 616s30 149.3 84.4 203.6C258.7 874 331.1 904 408 904s149.3-30 203.6-84.4C666 765.3 696 692.9 696 616c0-64.1-20.8-124.9-59.2-174.9L836 241.9V402c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V150c0-16.5-13.5-30-30-30zM408 828c-116.9 0-212-95.1-212-212s95.1-212 212-212 212 95.1 212 212-95.1 212-212 212z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="ant-descriptions-view"
              >
                <table>
                  <tbody>
                    <tr
                      class="ant-descriptions-row"
                    >
                      <td
                        class="ant-descriptions-item"
                        colspan="1"
                      >
                        <div
                          class="ant-descriptions-item-container"
                        >
                          <span
                            class="ant-descriptions-item-label"
                          >
                            姓名
                          </span>
                          <span
                            class="ant-descriptions-item-content"
                          >
                            name
                          </span>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-descriptions-row"
                    >
                      <td
                        class="ant-descriptions-item"
                        colspan="1"
                      >
                        <div
                          class="ant-descriptions-item-container"
                        >
                          <span
                            class="ant-descriptions-item-label"
                          >
                            手机
                          </span>
                          <span
                            class="ant-descriptions-item-content"
                          >
                            mobile
                             
                            <div
                              class="css-acbihw"
                            >
                              <span
                                aria-label="check-circle"
                                class="anticon anticon-check-circle"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="check-circle"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                    fill="#52c41a"
                                  />
                                  <path
                                    d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                                    fill="#f6ffed"
                                  />
                                  <path
                                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                                    fill="#52c41a"
                                  />
                                </svg>
                              </span>
                            </div>
                          </span>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-descriptions-row"
                    >
                      <td
                        class="ant-descriptions-item"
                        colspan="1"
                      >
                        <div
                          class="ant-descriptions-item-container"
                        >
                          <span
                            class="ant-descriptions-item-label"
                          >
                            地区
                          </span>
                          <span
                            class="ant-descriptions-item-content"
                          >
                            country province city
                          </span>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
